sdk/core/Azure.Core/src/Shared/GeoRedundantReadFallbackPolicy.cs

```diff
@@ -0,0 +1,134 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core.Pipeline;
+
+namespace Azure.Core.Shared
+{
+    /// <summary>
+    ///
+    /// </summary>
+    internal class GeoRedundantReadFallbackPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly string[] _readFallbackHosts;
+        private int? _fallbackIndex;
```

Comment:
you should use int/long and just treat -1 as null

Reply:
Yep, just realized that this will be necessary to use Interlocked.